---
title: "Market Analysis and Patient Sub-segment Identification for Glioblastoma Multiforme (GBM) Entry Strategy"
author: "Presented by Xuan Lu"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This report offers a comprehensive analysis of the Glioblastoma Multiforme (GBM) therapeutic area to facilitate our client's decision-making process as they contemplate entering this specialized field. The primary focus is on understanding the structure of the GBM market, specifically identifying clinically significant patient subgroups that receive distinct treatments. The ultimate objective is to unveil distinct opportunities within this challenging and critical medical landscape.

## GBM Overview

Glioblastoma Multiforme (GBM) is a common and very aggressive type of brain cancer in adults. It starts in the glial cells of the brain, mostly from astrocytes. GBM grows quickly, spreads fast, and often comes back after treatment, making it difficult to treat. Common treatments for GBM include surgery, radiation, and chemotherapy. Temozolomide, for example, is one of the most commonly used chemotherapy. As many as the treatments exist, prognosis of GBM remains fairly low-- according to American Cancer Society, the 5-year relative survival rate of GBM is 22% for adults aged 20-44, 9% for aged 45-54, and 6% for aged 55-64.

#### Descriptive Statistics

```{r data import and cleaning, message=FALSE, warning=FALSE, echo=FALSE, results='hide'}
# Load the packages
library(dplyr)

# Read the dataset
gbm <- read.csv("GBM patient data.csv") |>
  janitor::clean_names()
```

To better understand the GBM market, we will do some analysis utilizing GBM data of `r n_distinct(gbm$patient_id)` patients from `r n_distinct(gbm$md_id)` healthcare providers.

```{r raw data explorations, message=FALSE, warning=FALSE, echo=FALSE, results='hide'}
library(summarytools)
summarytools::dfSummary(gbm) %>% knitr::kable()
```

```{r data cleaning, message=FALSE, warning=FALSE, echo=FALSE, results='hide'}
# Identify variables with names starting with "comorbidity_"
comorbidity_vars <- grep("^comorbidity_", names(gbm), value = TRUE)

# Filter out numeric variables from comorbidity variables (binary, 1 = yes and 0 = no)
comorbidity_num_vars <- comorbidity_vars[sapply(gbm[comorbidity_vars], is.numeric)]

# Load necessary libraries
library(dplyr)
library(forcats)

# Renaming variables
gbm <- gbm %>%
  rename(
          patient_level_of_involvement = patient_s_level_of_involvement_in_gbm, 
          comorbidity_alzheimer = comorbidity_alzheimer_s, 
          comorbidity_other_cancer_notgbm = comorbidity_other_cancer_besides_gbm, 
          patient_travel_time_lt30min = travel_time_to_your_office_30_min, 
          patient_treatment_goals = patient_s_treatment_goals, 
          pct_of_tumor_mass_surgically_resected = x_of_tumor_mass_surgically_resected
         )
gbm <- gbm %>%
  # Recode "comorbidity_" numeric variables as "No", "Yes", or "Unknown"
  mutate(across(starts_with("comorbidity_"), ~ recode_factor(.,
                                                             `0` = "No",
                                                             `1` = "Yes",
                                                             `9` = "Unknown"))) %>%
  # Recode specified variables based on criteria
  mutate(across(all_of(c(
    "adequate_caretaker_support", 
    "patient_travel_time_lt30min", 
    "mgmt_methylated", 
    "egfr_mutated", 
    "tp53_mutated", 
    "idh1_idh2_mutated",
    "pd_l1_overexpressed")), 
    ~ recode_factor(.,
                    `0` = "No",
                    `1` = "Yes",
                    `9` = "Unknown"))) %>%
  # Recode ECOG variables with specific criteria
  mutate(across(all_of(c("ecog_at_1st_line", "ecog_at_2nd_line")), 
                ~ recode_factor(.,
                                `0` = "ECOG 0",
                                `1` = "ECOG 1",
                                `2` = "ECOG 2",
                                `3` = "ECOG 3",
                                `4` = "ECOG 4",
                                `9` = "Unknown"))) %>%
  # Recode character variables based on criteria
  mutate(across(all_of(c("regimen_in_2nd_line", "primary_insurance")),
                ~ ifelse(. == "999" | . == "9", "Unknown", .))) %>%
  mutate(pct_of_tumor_mass_surgically_resected = ifelse(
    pct_of_tumor_mass_surgically_resected == 999, NA, pct_of_tumor_mass_surgically_resected)) %>%
  # Recode involvement and treatment goals
  mutate(patient_level_of_involvement = recode(patient_level_of_involvement, 
                                               "Active / knowledgeable / questioning" = "High", 
                                               "Average level of engagement with the disease" = "Average", 
                                               "Passive / not interested / unengaged" = "Low"),
         patient_treatment_goals = recode(patient_treatment_goals, 
                                          "More focused on Survival" = "Survival",
                                          "More focused on Quality of Life" = "Life Quality"),
         patient_level_of_involvement = factor(patient_level_of_involvement, 
                                               levels = c("Low", "Average", "High"), 
                                               ordered = TRUE),
         patient_treatment_goals = factor(patient_treatment_goals, 
                                          levels = c("Survival", "Life Quality"), 
                                          ordered = TRUE), 
         race = recode(race, 
                       "Black/African" = "Black", 
                       "White/Caucasian" = "White",
                       "Spanish/Hispanic/Latino" = "Hispanic"))

# Create a function to concatenate comorbidity conditions
combine_comorbidities <- function(row) {
  comorbidities <- c()
  
  # List of all comorbidity variables
  comorbidity_vars <- c("comorbidity_renal_impairment", "comorbidity_anemia", 
                        "comorbidity_copd_pulmonary", "comorbidity_diabetes", 
                        "comorbidity_chf_cong_heart_failure", "comorbidity_neuropathies", 
                        "comorbidity_other_neurological", "comorbidity_hepatic_insufficiency", 
                        "comorbidity_other_cytopenia", "comorbidity_autoimmune_disorder", 
                        "comorbidity_hypertension", "comorbidity_other_cardiovascular", 
                        "comorbidity_alzheimer", "comorbidity_other_cancer_notgbm")
  
  # Loop through comorbidity variables
  for (var in comorbidity_vars) {
    if (row[[var]] == "Yes") {
      # Add the condition name to comorbidities list, removing 'comorbidity_' prefix and replacing '_' with ' '
      comorbidities <- c(comorbidities, gsub("_", " ", sub("comorbidity_", "", var)))
    }
  }
  
  # Add non-blank values from 'comorbidity_other_specify_1'
  if (row[["comorbidity_other_specify_1"]] != " ") {
    comorbidities <- c(comorbidities, row[["comorbidity_other_specify_1"]])
  }
  
  # Combine all comorbidities into a single string, separated by commas
  paste(comorbidities, collapse = ", ")
}

# Apply the function to each row of the data frame and create a new variable
gbm$combined_comorbidities <- apply(gbm, 1, combine_comorbidities)

# Create a function to count the number of comorbidities
count_comorbidities <- function(row) {
  comorbidity_count <- 0
  
  # List of all comorbidity variables
  comorbidity_vars <- c("comorbidity_renal_impairment", "comorbidity_anemia", 
                        "comorbidity_copd_pulmonary", "comorbidity_diabetes", 
                        "comorbidity_chf_cong_heart_failure", "comorbidity_neuropathies", 
                        "comorbidity_other_neurological", "comorbidity_hepatic_insufficiency", 
                        "comorbidity_other_cytopenia", "comorbidity_autoimmune_disorder", 
                        "comorbidity_hypertension", "comorbidity_other_cardiovascular", 
                        "comorbidity_alzheimer", "comorbidity_other_cancer_notgbm")
  
  # Loop through comorbidity variables
  for (var in comorbidity_vars) {
    if (row[[var]] == "Yes") {
      comorbidity_count <- comorbidity_count + 1
    }
  }
  
  # Check 'comorbidity_other_specify_1' variable
  if (row[["comorbidity_other_specify_1"]] != " ") {
    comorbidity_count <- comorbidity_count + 1
  }
  
  return(comorbidity_count)
}

# Apply the function to each row of the data frame and create a new variable
gbm$comorbidity_count <- apply(gbm, 1, count_comorbidities)
```

```{r Descriptive Statistics, message=FALSE, warning=FALSE, echo=FALSE}
# Load necessary libraries
library(dplyr)
library(tidyverse)
library(psych)
library(broom)

# 1. Calculate summary statistics for continuous variables:

# Select the variables to compute descriptive statistics
numeric_vars <- gbm %>%
  select_if(is.numeric) %>%
  select(-md_id, -patient_id)

# Calculate summary statistics
summary_stats <- describe(numeric_vars) %>% select(mean, sd, min, median, max, range)

# Print the statistic table
summary_stats %>% knitr::kable()

# 2. Generate frequency tables for categorical variables

# Define the categorical variables you want to analyze
categorical_vars <- gbm %>%
  select_if(~ is.character(.) || is.factor(.))

# Create an empty data frame to store the results
result_freq <- data.frame(variable = character(0), categories = character(0), count = numeric(0), percent = character(0))

# Loop through the categorical variables
for (var in colnames(categorical_vars)) {
  cat_summary <- gbm %>% 
    group_by(!!sym(var)) %>%
    summarize(count = n()) %>%
    mutate(percent = scales::percent(count / sum(count), scale = 100))
  # Add the variable name to the results
  cat_summary <- cat_summary %>%
    mutate(variable = var)
  # Rename the categories column
  names(cat_summary)[names(cat_summary) == var] <- "categories"
  # Select and arrange the columns
  cat_summary <- cat_summary %>%
    select(variable, categories, count, percent)
  # Append to the result data frame
  result_freq <- bind_rows(result_freq, cat_summary)
}

cat_vars_X2 <- select(categorical_vars, c("race", "patient_level_of_involvement"))

# Create an empty data frame to store the results
chi_square_results <- data.frame(variable = character(0), chi_square = numeric(0), p_value = numeric(0))

# Loop through each categorical variable
for (var in colnames(cat_vars_X2)) {
  # Subset the data for the current variable
  var_data <- result_freq %>%
    filter(variable == var)
  # Create a contingency table
  contingency_table <- table(var_data$categories, var_data$count)
  # Perform chi-square test
  chi_square_test <- chisq.test(contingency_table)
  # Extract chi-square and p-value
  chi_square_value <- chi_square_test$statistic
  p_value <- chi_square_test$p.value
  # Add results to the chi_square_results data frame
  chi_square_results <- rbind(chi_square_results, data.frame(variable = var, chi_square = chi_square_value, p_value = p_value))
}

# Print the tables
result_freq %>% 
  filter(variable %in% c("gender", "race", "patient_level_of_involvement", "patient_treatment_goals", "line_of_therapy")) %>%
  knitr::kable()
# Chi-square test did not find any significant findings; do not show the table. 
# print("Selected categorical variables, Chi-square test:")
# chi_square_results %>% 
#  knitr::kable()

# 3. Generate frequency table for comorbidities

# Split the combined_comorbidities column into individual comorbidities
comorbidities_list <- unlist(strsplit(as.character(gbm$combined_comorbidities), ", "))
# Calculate the frequency of each comorbidity
comorbidity_frequency <- table(comorbidities_list)
# Convert the table to a data frame
comorbidity_df <- as.data.frame(comorbidity_frequency)
# Rename the columns for clarity
colnames(comorbidity_df) <- c("Comorbidity", "Frequency")
# Order the data frame by frequency in descending order
comorbidity_df <- comorbidity_df[order(-comorbidity_df$Frequency), ]

# List of objects to keep
objects_to_keep <- c("gbm", "summary_stats", "result_freq", "comorbidity_df")
# Get a list of all objects in the current environment
all_objects <- ls()
# Find objects to remove
objects_to_remove <- setdiff(all_objects, objects_to_keep)
# Remove objects to be deleted
rm(list = objects_to_remove)
```

In this dataset, the age at GBM diagnosis varies, with the median and mean closely aligned, suggesting a **symmetric distribution**. GBM primarily affects **adults**, with no diagnoses on or before age 18. Surgical resection has a mean percentage of 70.1% and a slightly higher median indicating a **left-skewed distribution**. Average comorbidity count is around 1, highlighting **minimal comorbidities**.

The majority of participants are **White** (75.1%) and **male** (65%). A more diverse study is needed for a comprehensive view of GBM across races. Most patients undergo **second-line therapy** with a balanced distribution between therapy lines. Around half have **average GBM involvement**, the remainder has a larger proportion showing high involvement compared to low involvement. Most patients prioritize **survival** (69%) over life quality (31%). No categorical variable displayed significant disparity in the chi-square test (significance level: 0.05).

#### Comorbidities

```{r 5 most common regimens, echo = FALSE, results='hide', message=FALSE, warning=FALSE}
# Display the data frame
comorbidity_df %>% 
  head(5) %>%
  knitr::kable()
```

There are a number of comorbidities that can co-occur with GBM. There is a maximum of `r max(gbm$comorbidity_count)` comorbidities co-occurring with GBM in record. **Hypertension** is the most prevalent comorbidity (24% of sample), followed by **diabetes** (18.4% of sample), **anemia** (15.6%), **chronic obstructive pulmonary disease** (11.5%), and **renal impairment** (10.5%), many of which share similar risk factors and medication overlap. 

#### Genetic Factors

```{r genetic factors, echo = FALSE, results='hide', message=FALSE, warning=FALSE}
# Create a function to concatenate genetic factors
combine_genetic <- function(row) {
  genetic_factors <- c()
  
  # List of all genetic variables
  genetic_vars <- c("mgmt_methylated", "egfr_mutated", 
                    "tp53_mutated", "idh1_idh2_mutated", 
                    "pd_l1_overexpressed")
  
  # Loop through genetic variables
  for (var in genetic_vars) {
    if (row[[var]] == "Yes") {
      # Add the condition name to genetic_factors list
      genetic_factors <- c(genetic_factors, gsub("_", " ", var))
    }
  }
  
  # Combine all genetic factors into a single string, separated by commas
  paste(genetic_factors, collapse = ", ")
}
```

```{r genetic table, echo = FALSE, message=FALSE, warning=FALSE}
# Apply the function to each row of the data frame and create a new variable
library(knitr)
genetic_vars <- c("mgmt_methylated", "egfr_mutated", 
                  "tp53_mutated", "idh1_idh2_mutated", 
                  "pd_l1_overexpressed")

combined_genetic <- sapply(gbm[genetic_vars], table)/nrow(gbm) * 100
combined_genetic %>% knitr::kable()
```

Table above shows genetic factor variables and their percentages. There are a number of genetic factors associated with GBM. In this dataset, genetic factors include: 

- **MGMT methylated:** The MGMT gene's methylation status. When methylated, the MGMT gene is less active, which can affect how GBM responds to certain treatments. 
- **EGFR mutated:** Indicates mutations in the EGFR gene, which plays a role in cell growth. Mutations can lead to uncontrolled cell growth, a characteristic of many tumors.
- **TP53 mutated:** Mutations in the TP53 gene are present. This gene helps regulate cell growth and repair damaged DNA. Mutations can promote tumor development.
- **IDH1/IDH2 mutated:** Mutations in the IDH1 or IDH2 genes are linked to cancer growth. These mutations can influence treatment choices and outcomes for GBM patients.
- **PD-L1 overexpressed:** The PD-L1 protein is overproduced, potentially suppressing the body's immune response to the tumor. This can inform decisions about immunotherapy treatments.

#### Lines of Therapy

Chi-squared test did not identify significant difference between the proportions of patients taking 1st line and 2nd line of therapy. For the distinct difference between 1st and 2nd line of therapies in nature, we will next explore some visualizations sub-grouped by line of therapy. 

```{r Most and least popular regimens, results='hide', echo=FALSE, message=FALSE, warning=FALSE}
# Calculate the most and least popular regimens for the 1st line of therapy
max_regimen_1st <- result_freq %>% 
  filter(variable == "regimen_in_1st_line") %>%
  arrange(desc(count)) %>%
  slice(1)

min_regimen_1st <- result_freq %>% 
  filter(variable == "regimen_in_1st_line") %>%
  arrange(count) %>%
  slice(1)

# Filter out "Unknown" and select regimens for the 2nd line of therapy
regimen_2nd_line_filtered <- result_freq %>% 
  filter(variable == "regimen_in_2nd_line", categories != "Unknown")

# Calculate the total count after excluding "Unknown"
total_count_filtered <- sum(regimen_2nd_line_filtered$count)

# Calculate the percentages
regimen_2nd_line_filtered <- regimen_2nd_line_filtered %>%
  mutate(percent = scales::percent(count / total_count_filtered, scale = 100))

# Identify the most and least popular regimens
max_regimen_2nd <- regimen_2nd_line_filtered %>% 
  arrange(desc(count)) %>%
  slice(1)

min_regimen_2nd <- regimen_2nd_line_filtered %>% 
  arrange(count) %>%
  slice(1)
```

```{r categorical_vars, echo=FALSE, message=FALSE, warning=FALSE, results='hide'}
categorical_vars <- gbm %>%
  select_if(is.character)
regimen_X2 <- select(categorical_vars, c("regimen_in_1st_line", "regimen_in_2nd_line"))

# Create an empty data frame to store the results
chi_square_results <- data.frame(variable = character(0), chi_square = numeric(0), p_value = numeric(0))

# Loop through each categorical variable
for (var in colnames(regimen_X2)) {
  # Subset the data for the current variable
  var_data <- result_freq %>%
    filter(variable == var)
  # Create a contingency table
  contingency_table <- table(var_data$categories, var_data$count)
  # Perform chi-square test
  chi_square_test <- chisq.test(contingency_table)
  # Extract chi-square and p-value
  chi_square_value <- chi_square_test$statistic
  p_value <- chi_square_test$p.value
  # Add results to the chi_square_results data frame
  chi_square_results <- rbind(chi_square_results, data.frame(variable = var, chi_square = chi_square_value, p_value = p_value))
}

# Output Table
print("Regimens, Chi-square test:")
chi_square_results %>% 
  knitr::kable()
```

```{r explorations, results='hide', echo=FALSE, message=FALSE, warning=FALSE}
summary(gbm)
sum(is.na(gbm))
# [1] 204

# Define your categorical variables
categorical_vars <- c("country", "line_of_therapy", "gender", "race", 
                      "adequate_caretaker_support", "patient_travel_time_lt30min", 
                      "patient_level_of_involvement", "patient_treatment_goals", 
                      "primary_insurance", "ecog_at_1st_line", 
                      "ecog_at_2nd_line", "mgmt_methylated", "egfr_mutated", 
                      "tp53_mutated", "idh1_idh2_mutated", "pd_l1_overexpressed", 
                      "regimen_in_1st_line", "regimen_in_2nd_line", "combined_comorbidities")

# Function to apply unique() to all categorical variables
apply_unique <- function(data, vars) {
  for (var in vars) {
    cat("Unique values for", var, ":\n")
    print(unique(data[[var]]))
    cat("\n")
  }
  
  # Apply unique() to variables starting with "Comorbidity_"
  comorbidity_vars <- names(data)[str_starts(names(data), "comorbidity_")]
  for (var in comorbidity_vars) {
    cat("Unique values for", var, ":\n")
    print(unique(data[[var]]))
    cat("\n")
  }
}

# Apply the function to your data
apply_unique(gbm, categorical_vars)
```

- **1st Line of Therapy:** Regimen used in 1st line of therapy include `r unique(result_freq$categories[result_freq$variable == "regimen_in_1st_line"])`. The most popular regimen used is `r max_regimen_1st$categories`, which accounts for `r max_regimen_1st$percent` of all regimens recorded. The least popular regimen used is `r min_regimen_1st$categories`, specifically `r unique(gbm$regimen_in_1st_line_other)`, which altogether accounts for `r min_regimen_1st$percent` of all regimens recorded. 
- **2nd Line of Therapy:** Regimen used in 2nd line of therapy include `r unique(result_freq$categories[result_freq$variable == "regimen_in_2nd_line"])`. Excluding Unknown, the most popular regimen used is `r max_regimen_2nd$categories`, which accounts for `r max_regimen_2nd$percent` of all regimens recorded. The least popular regimen used is `r min_regimen_2nd$categories`, which accounts for `r min_regimen_2nd$percent` of all regimens recorded.

##### ECOG

The Eastern Cooperative Oncology Group (ECOG) Performance Status is a widely-accepted scale utilized to evaluate the functional capacity of cancer patients. This assessment tool offers insights into a patient's ability to endure treatment and their overall prognosis. A lower score denotes a better functional status, whereas a higher score suggests more significant functional impairment.

```{r ECOG table, echo = FALSE, message=FALSE, warning=FALSE}
# Apply the function to each row of the data frame and create a new variable
ecog_list <- c("ecog_at_1st_line", "ecog_at_2nd_line")

# Convert factors to character temporarily
gbm[ecog_list] <- lapply(gbm[ecog_list], as.character)

# Get a list of tables
ecog_tables <- lapply(gbm[ecog_list], table)

# Convert the counts to percentages
ecog_percent <- lapply(ecog_tables, function(tbl) {
  tbl_percent <- (tbl / nrow(gbm)) * 100
  return(tbl_percent)
})

# Bind them together into a data frame for better presentation
ecog_df <- data.frame(do.call(cbind, ecog_percent))

# Display the table
knitr::kable(ecog_df)
```

- ECOG 0: Fully active, able to carry on all activities without restriction. This subsegment of patients is generally in the best health amongst those with cancer and is likely to tolerate aggressive treatments well.
- ECOG 1: Restricted in physically strenuous activity but able to walk and carry out light work. These individuals might require some adjustments to their treatment regimens but are largely capable of leading an active life.
- ECOG 2: Ambulatory and capable of self-care, but unable to work; out of bed more than 50% of waking hours. The health of this segment is moderately compromised, warranting careful consideration of treatment intensity to prevent further functional decline.
- ECOG 3: Capable of only limited self-care, confined to bed or chair more than 50% of waking hours. Such patients are significantly impaired and might benefit from supportive care interventions to enhance their quality of life.
- ECOG 4: Completely disabled, cannot carry out any self-care, confined to bed or chair all the time. This is the most critically impaired group. Palliative care and symptom management might be of prime importance to these patients.

For those on 1st line of treatment, ECOG 1 accounts for an overwhelming proportion, followed by ECOG 0, ECOG 2, and ECOG 4. This distribution suggests a mix of patients in terms of health status during the 1st line of treatment.

For those on 2nd line of treatment, ECOG 2 and 3 relatively evenly split the predominant proportions of patients, indicating that many of these patients have moderate functional limitations. This is followed by ECOG 0 and ECOG 3. This pattern suggests that as patients progress to the 2nd line of treatment, there is a broader spread of functional status, with a noticeable number of patients in both moderate and more severe categories.

## Visualizations - Demographics, Treatment Line, Patient Involvement, Treatment Goals

In strategizing market entry, understanding patients' involvement levels and their treatment objectives is pivotal for anticipating demands and pinpointing potential therapeutic niches. 

```{r Visualizations: line of therapies, echo=FALSE, fig.width=8, fig.height=3}
library(ggplot2)
line1_filtered <- gbm %>% filter(line_of_therapy == "1L")
line2_filtered <- gbm %>% filter(line_of_therapy == "2L")

line_labs <- c("1st Line of Therapy", "2nd Line of Therapy")
names(line_labs) <- c("1L", "2L")

ggplot(gbm, aes(x = gender, y = age_at_diagnosis)) + 
  geom_boxplot(outlier.colour="black", outlier.shape=16,
             outlier.size=2, notch=FALSE) + 
  facet_grid(. ~ line_of_therapy, labeller = labeller(line_of_therapy = line_labs)) +
  labs(
    title = "Box Plot: Distribution of Age at Diagnosis by Gender",
       x = "Gender",
       y = "Age at Diagnosis"
      ) + 
  theme(
    title = element_text(size=8),
    axis.text.x = element_text(size=8), 
    axis.text.y = element_text(size=8),
    strip.text = element_text(size = 8), 
    legend.position = "bottom")

ggplot(gbm, aes(x = patient_level_of_involvement, fill = primary_insurance)) +
  geom_bar(position = "stack") +
  facet_grid(. ~ line_of_therapy, labeller = labeller(line_of_therapy = line_labs)) + 
  labs(title = "Bar Plot of Patient Involvement by Primary Insurance", x = "Patient Involvement Level", s = "Count") + 
  theme(
    title = element_text(size=8),
    axis.text.x = element_text(size=8), 
    axis.text.y = element_text(size=8),
    strip.text = element_text(size = 8), 
    legend.position = "right") + 
  viridis::scale_fill_viridis(discrete = TRUE)

ggplot(gbm, aes(x = patient_treatment_goals, fill = primary_insurance)) +
  geom_bar(position = "stack") +
  facet_grid(. ~ line_of_therapy, labeller = labeller(line_of_therapy = line_labs)) + 
  labs(title = "Bar Plot of Patient Treatment Goals by Primary Insurance", x = "Patient Treatment Goal", s = "Count") + 
  theme(
    title = element_text(size=8),
    axis.text.x = element_text(size=8), 
    axis.text.y = element_text(size=8),
    strip.text = element_text(size = 8),
    legend.position = "right") + 
  viridis::scale_fill_viridis(discrete = TRUE)
```

As indicated by the provided graphs:

- **Age at Diagnosis:** This parameter remains consistent irrespective of therapy lines and gender, suggesting that age might not be a primary differentiator in this market.
- **Primary Insurance Dynamics:** While the choice of primary insurance is influenced by patients' involvement levels and treatment goals, it remains uniform across different therapy lines.
  - For patients with medium to high involvement levels and a primary focus on survival, HMO, PPO, and Medicare are the prevailing insurance choices.
  - Conversely, Medicare emerges as the dominant insurance for those with low involvement levels, particularly for patients prioritizing quality of life.
- **Socioeconomic Implications:** The types of primary insurance often serve as proxies for patients' socioeconomic backgrounds. For instance:
  - Medicaid typically caters to a younger demographic, specifically those under 65 with constrained financial resources.
  - Medicare predominantly targets the senior population aged 65 and above.

In light of these insights, it's evident that insurance dynamics, potentially steered by demographic factors, play a crucial role in this market. Recognizing and leveraging these variations will be instrumental in fine-tuning our client's market segmentation strategy.

## Key Findings and Points to Consider: 

1. **Patient Demographics:** GBM primarily affects adults, with a mean age at diagnosis of approximately 57 years.
2. **Treatment Regimens:** TMZ mono is the most popular regimen for the 1st line of therapy. Avastin mono is the preferred choice for the 2nd line of therapy.
3. **Genetic Factors:** Multiple genetic mutations influence treatment decisions, with a significant portion of patients having unknown genetic statuses.
4. **Patient Insurance and Socioeconomic Background:** Types of primary insurance, including Medicaid and Medicare, offer insights into the patients' socioeconomic backgrounds.

## Opportunity Analysis:

1. **Diverse Study Requirement:** Since 75.1% of participants are White, there's an opportunity to focus on more diverse clinical studies to provide a comprehensive view of GBM across races.
2. **Genetic Mutation Treatments:** Given the significant proportion of patients with unknown genetic statuses, there's potential in offering more genetic testing or therapies that cater to known mutations.
3. **Addressing the 2nd Line of Therapy:** With Avastin mono being the preferred choice, new drug combinations or therapeutic solutions could be researched for 2nd line treatments.
4. **Younger Demographic Engagement:** With Medicaid catering to a younger demographic under 65 with constrained financial resources, there's an opportunity to offer specialized programs or financial aid to this segment.

## Recommendations:

1. **Diversify Clinical Trials:** The client should consider diversifying clinical trials, ensuring broader racial and ethnic representation, thus allowing for treatments that cater to a broader patient base.
2. **Expand Genetic Testing:** Advocate for and possibly offer genetic testing services for patients to determine mutation statuses. This will help in personalizing treatments more effectively.
3. **Engage Insurance Providers:** Build partnerships with insurance providers, especially Medicaid and Medicare, to potentially offer more affordable treatment options or payment plans.
4. **Patient Education:** Develop and implement educational programs, particularly targeting the younger demographic and those with low involvement levels. This can aid in early diagnosis and treatment adherence.
5. **Therapeutic R&D:** Invest in research and development for treatments targeting specific genetic mutations like MGMT methylated or EGFR mutated.
6. **Socioeconomic Support Programs:** Establish support programs for patients under Medicaid, focusing on financial aid, counseling, and other support services.

